lib/SuiteCRM/API/JsonApi/v1/Resource/Relationship.php

```diff
@@ -141,4 +125,35 @@ public function toJsonApiResponse() {
         }
         return $payload;
     }
-}
\ No newline at end of file
+
+    /**
+     * Sets the link property up
+     * @parm Resource|ResourceIdentiifer
+     */
+    private function withResourceObject($related)
+    {
+        if($this->getType() === null) {
+            $this->type = $related->getType();
+        } elseif ($this->getType() !== $related->getType()) {
+            throw new Forbidden('[Relationship] [Incompatible Resource Type] "'. $related->getType().'"');
+        }
+
+        $this->id = $related->getId();
+
+        if ($this->relationshipType === RelationshipType::TO_ONE) {
+            $this->link = $related;
+        } elseif ($this->relationshipType === RelationshipType::TO_MANY) {
+            $this->link[] = $related;
+        } else {
+            throw new ApiException('[Relationship] [Unsupported Relationship Type]');
```

Comment:
Can you add the relationshipType into the error message as at other places, pls

Reply:
sure. done. :+1: